core/blockchain.go

```diff
@@ -942,7 +944,6 @@ type WriteStatus byte
 const (
 	NonStatTy WriteStatus = iota
 	CanonStatTy
```

Comment:
I'm just wondering, what is the meaning of `NonStatTy` and `CanonStatTy`?

Reply:
CanonStatTy is that the block added is part of the cannonical chain
NonStatTy is used for the blocks that had some error
After this PR there was the SideStatTy and even after, a SplitStatTy

I could remove both, and just change the state for a boolean (valid or not) but I thought that it would be easier to follow, thinking in the merges from upstream